### PR TITLE
Dependency Cleanup

### PR DESCRIPTION
**Pull request by mixonic**
* ~Add `ember-cli-shims` as a dependency. This allows you to use this library as a dependency even if the main app does not depend on `ember-cli-shims`.~
* Upgrade Ember-CLI
* Expand ember-try to cover Ember 2.8 and 2.12 LTS releases.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,50 +2,52 @@
...
-  },
-  "engines": {
-    "node": ">= 4"
+  "dependencies": {
+    "@glimmer/resolver": "^0.4.1",
+    "babel-plugin-debug-macros": "^0.1.10",
+    "broccoli-funnel": "^1.1.0",
+    "broccoli-merge-trees": "^2.0.0",
+    "ember-cli-babel": "^6.3.0",
+    "ember-cli-shims": "^1.1.0",
```
Comment (rwjblue) [inline package.json]: I'm pretty sure that we don't actually want this. Why did you need to add it?

Comment (mixonic) [inline package.json]: Yeah this is the main thrust here. I should split them out but was too lazy to rebase w/ lockfile :-p I'll split the updates out.

Basically with the new modules, you would expect that you can now *stop* using ember-cli-shims. Why would you need both the module API provided by ember/ember-cli-babel and the legacy shims as well?

However once you drop ember-cli-shims as an app dependency, nothing can `import Ember from 'ember'`. So any addons that don't have `ember-cli-shims` as a dependency fail to have that import.

There are a few ways to fix this. Open to another idea.

I'm not clear on what the best practices for addons are in the new modules world. You would expect addon authors to want to write addons with the new APIs, but obviously continue to support older versions of Ember.

Comment (rwjblue) [inline package.json]: Addons can use the new modules in their addon/ (and addon-test-support) tree as long as they themselves have ember-cli-babel@6.6.0 or higher as a dependency. 

Since an addon does not control the mechanism for transpilation of the app/ and test-support/ trees (these are compiled by the app's own preprocessor chain) an addon should _only_ reexport from their addon tree into app/ and from their addon-test-support tree in their test-support tree. 

To the specific point RE: the 'ember' module itself (which is not currently provided by ember-cli-babel@6.6.0), we just need to fix this and ensure that it is. We definitely do not want to add ember-cli-shims as a dependency to any addons.

**Comment by mixonic** [comment]
Great, after some work with @rwjblue the ember-cli-shims stuff is no longer required. Instead, ember-cli-babel 6.8.0 provides `import Ember from 'ember'`.

At this point, the diff here only represents an upgrade of dependencies and expansion of ember-try coverage.

**Comment by mixonic** [comment]
@rwjblue good to go.
